java/archive.gradle

```diff
@@ -1,55 +1,51 @@
-apply plugin: 'maven'
+apply plugin: 'maven-publish'
 apply plugin: 'signing'
 
-artifacts {
-    archives javadocJar, sourcesJar
-}
-
-signing {
-    sign configurations.archives
-}
-
-uploadArchives {
-    repositories {
-        mavenDeployer {
-            beforeDeployment { MavenDeployment deployment -> signing.signPom(deployment) }
-
-            repository(url: "https://oss.sonatype.org/service/local/staging/deploy/maven2/") {
-                authentication(userName: ossrhUsername, password: ossrhPassword)
-            }
-
-            snapshotRepository(url: "https://oss.sonatype.org/content/repositories/snapshots/") {
-                authentication(userName: ossrhUsername, password: ossrhPassword)
-            }
-
-            pom.project {
-                name 'Scalar Admin'
-                packaging 'jar'
-                // optionally artifactId can be defined here 
-                description 'An admin interface and tool for Scalar services.'
-                url 'https://github.com/scalar-labs/scalar-admin'
-
-                scm {
-                    connection 'scm:git:https://github.com/scalar-labs/scalar-admin.git'
-                    developerConnection 'scm:git:https://github.com/scalar-labs/scalar-admin.git'
-                    url 'https://github.com/scalar-labs/scalar-admin'
-                }
-
+publishing {
+    publications {
+        mavenJava(MavenPublication) {
+            artifactId = 'scalar-admin'
+            artifact jar
+            artifact javadocJar
+            artifact sourcesJar
```

Comment:
To not publish the shadowJar file (scalar-admin-<version>-all.jar), I set 
```
artifact jar
artifact javadocJar
artifact sourcesJar
```

instead of
```
from components.java
artifact javadocJar
artifact sourcesJar
```

because components.java contains all *.jar files including shadowJar as well.

I tried to use the same configuration in scalardb/schema-loader/build.gradle
```
    // not to publish the fat jar
    project.gradle.startParameter.excludedTaskNames= ["shadowJar"]
```

it's able to skip the shadowJar task but seems not to exclude the shadowJar file from the conponents.java, so I choose the `artifact jar` solution.

Reply:
It seems like it's the best solution 👍  I'll do the same in the scalardb project later.

Reply:
Nice!